src/core_modules/capture-core/utils/navigateToTrackedEntityDashboard.js

```diff
@@ -1,7 +1,21 @@
 import { config } from 'd2';
 
-export const navigateToTrackedEntityDashboard = (teiId, orgUnitId, scopeSearchParam) => {
+export const navigateToTrackedEntityDashboard = (teiId, orgUnitId, scopeSearchParam, currentUrl) => {
     const { baseUrl } = config;
     const instanceBaseUrl = baseUrl.split('/api')[0];
-    window.location.href = `${instanceBaseUrl}/dhis-web-tracker-capture/#/dashboard?tei=${teiId}&ou=${orgUnitId}&${scopeSearchParam}`;
+
+    const returnUrl =
+      // when developing locally the following will always be false
+      window.location.href.includes(instanceBaseUrl)
+          ?
+          window.location.href.split(instanceBaseUrl)[1]
+          :
+          `/dhis-web-capture/#${currentUrl}`;
```

Comment:
I dont think this is self explanatory but when you develop locally you can match the `instanceBaseUrl` (which will be somthing between these lines`https://debug.dhis2.org/ca-2.36`) with the current `window.location.href`. 

For this cases I hardcoded the value ``/dhis-web-capture/#${currentUrl}`;`

I think there is much room for improvement here but cant find a better solution. Very open to suggestions